pkg/component/status/status.go

```diff
@@ -0,0 +1,89 @@
+package status
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/k0sproject/k0s/internal/util"
+	"github.com/k0sproject/k0s/pkg/install"
+	"github.com/sirupsen/logrus"
+)
+
+type Status struct {
+	StatusInformation install.K0sStatus
+	L                 *logrus.Entry
+	address           string
+	httpserver        http.Server
+	listener          net.Listener
+}
+
+// Healthy dummy implementation
+func (s *Status) Healthy() error { return nil }
+
+// Init initializes component
+func (s *Status) Init() error {
+	s.L = logrus.WithFields(logrus.Fields{"component": "status"})
+
+	if len(s.address) == 0 {
+		s.address = path.Join(s.StatusInformation.K0sVars.RunDir, fmt.Sprintf("%s.sock", s.StatusInformation.Role))
```

Comment:
To determine the role of an instance via the socket, you first need to know the role of the instance so you know which socket to communicate with.

Reply:
We query both known sockets and gather data if present.